### PR TITLE
E2E changes for main branch

### PR DESCRIPTION
**Pull request by android-dev-lxl**

### CODE REVIEW

**Review by johnshea — APPROVED** [review]
LGTM with some small changes.
- Upgrade to latest versions
- Check if `@file:Suppress("UnstableApiUsage")` can be removed from `build.gradle.kts (:app)` after upgrading versions
- In file `GameScreen.kt`, in function `GameScreenPreview`, it should be using a `Surface` so it matches how `MainActivity.kt` is coded.

**Review by johnshea — CHANGES_REQUESTED** [review]
Nav bar is not translucent but should be.

Screenshot 2023-09-06 at 3 45 49 PM

**Comment by johnshea** [comment]
Should the app name "Unscramble" be overlapping with the camera?

Screenshot 2023-09-11 at 12 08 57 PM

Should the dialog be showing under the status bar?

Screenshot 2023-09-11 at 12 05 51 PM

**Comment by android-dev-lxl** [comment]
> Should the app name "Unscramble" be overlapping with the camera?
> 
> Screenshot 2023-09-11 at 12 08 57 PM
> Should the dialog be showing under the status bar?
> 
> Screenshot 2023-09-11 at 12 05 51 PM

I am not able to reproduce either.
image


image

**Comment by johnshea** [comment]
re: the dialog showing under status bar

To recreate - Use emulator "Pixel 6 API 33 w/ gesture navigation", turn emulator to landscape, click in the text edit box to enter your word

**Review by johnshea — CHANGES_REQUESTED** [review]
UI for edge-to-edge does not match the UI from branch `main` but it should.
Screenshot 2023-09-18 at 2 45 49 PM

File: app/src/main/java/com/example/unscramble/ui/theme/Theme.kt [hunk 1]
```diff
@@ -18,6 +18,7 @@ package com.example.unscramble.ui.theme
 
 import android.app.Activity
```
Comment (johnshea) [inline app/src/main/java/com/example/unscramble/ui/theme/Theme.kt:19]: Removed unused imports.

File: app/src/main/java/com/example/unscramble/ui/theme/Theme.kt
Comment (johnshea) [inline app/src/main/java/com/example/unscramble/ui/theme/Theme.kt:1]: nit: line 100 - remove extra blank line.

File: app/src/main/java/com/example/unscramble/MainActivity.kt [hunk 2]
```diff
@@ -30,11 +30,10 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
+            enableEdgeToEdge()
```
Comment (johnshea) [inline app/src/main/java/com/example/unscramble/MainActivity.kt:33]: E2e api call is in the wrong spot. It should be before `super.onCreate()`

**Review by johnshea — CHANGES_REQUESTED** [review]

**Review by johnshea — APPROVED** [review]
lgtm

**Pull request merged**